src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<? extends Machine> bulldozers = new BulldozerProducer().get();
+            return new ArrayList<>(bulldozers);
+        } else if (type == Excavator.class) {
+            List<? extends Machine> excavators = new ExcavatorProducer().get();
+            return new ArrayList<>(excavators);
+        } else if (type == Truck.class) {
+            List<? extends Machine> trucks = new TruckProducer().get();
+            return new ArrayList<>(trucks);
+        }
+        return null;
```

Comment:
return Collections.EMPTY_LIST instead. Returning null will force a user to add null checks or he may receive NPE